### ISSUE TITLE
Shields of Pakistan

### ISSUE THREAD

**Issue by bgo-eiu** [ZeLonewolf/openstreetmap-americana#396]
I am creating this issue to start with seeing as there are issues for various country shields, but intend to look into the shields more, as well as how road routes are mapped in Pakistan, then will make a pull request at some point.

Pakistan has a network of National Highway routes designated by the national government which connect the country's major cities, and a Motorway network of routes installed relatively recently with the intent of providing connections between Pakistan's Arabian Sea ports and the Khunjerab Pass at the border with China as part of the "China-Pakistan Belt Road Initiative."

National Highway `ref` designations are prefixed with `N-` and are displayed on a blue hexagonal shield:
![N-10 Pakistan National Highway Shield](https://upload.wikimedia.org/wikipedia/commons/thumb/6/61/Pakistan_N-10.svg/750px-Pakistan_N-10.svg.png)

On guide signage they may look like this:
![N-5 Shield on Guide Signage in Pakistan](https://upload.wikimedia.org/wikipedia/commons/4/4f/Road_sign_in_Islamabad.JPG)

The hexagon outline is not always shown; as you can see on this sign, N-35 (Karakoram Highway) is referenced without the hexagon shape (not specific to the route or carrying any different meaning, just omitted on some signs):
![N-35 Reference Number on Guide Signage in Pakistan](https://upload.wikimedia.org/wikipedia/commons/0/0b/Sign_for_foreigners_Karakoram_Highway_Sep_2004.jpg)

Motorways in Pakistan have `ref` designations prefixed with `M-` and green semi-circular shields:
![M-10 Pakistan Motorway Shield](https://upload.wikimedia.org/wikipedia/commons/thumb/2/29/Pakistan_M-10.svg/550px-Pakistan_M-10.svg.png)

They are often shown on matching green guide signage like this:
![M-2 on Guide Signage in Pakistan](https://upload.wikimedia.org/wikipedia/commons/1/13/M2_Sargodha.jpg)

Pakistan's provinces and territories also each have provincial or territorial highway networks. Their respective subdivisions assign different systems of `ref` designations for each, some of which do have their own shield designs but not all. The available images of signage for these is sporadic compared to the National Highways so it can be harder to discern, but here is an example a shield for the L-20 route, named Lahore Ring Road, of Punjab's provincial highway system:
![L-20 Lahore Ring Road, Punjab, Pakistan Shield](https://upload.wikimedia.org/wikipedia/commons/f/f9/L-20_Logo.png)

The same on guide signage (poor lighting in this picture unfortunately):
![L-20 Lahore Ring Road on Guide Signage in Punjab, Pakistan](File:Lahore Ring Road (L-20).jpg)
I am not sure if any other routes use the `L-` prefix; it takes the first letter of the city name despite being designated and maintained by the province of Punjab rather than the municipality of Lahore.

Further, there is an additional highway network known as the Strategic Highway network which is maintained and operated by the Pakistan Ministry of Defence. These have their own `ref` designations with the `S-` prefix followed by a number. I am not yet sure if these have their own shields. Unhelpfully, both the provinces of Sindh and Khyber-Pakhtunkhwa also use `S` as a prefix. `S-2` (Strategic Highway), `S-2` (Khyber-Pakhtunkhwa Provincial Highway, and `S02` (Sindh Provincial Highway) are all different routes.

Some of Pakistan's National Highways and Motorways are part of the Asian Highway Network discussed in issue #122. At the moment I am unsure of the extent to which Asian Highway shields are signed in Pakistan, and how they are styled compared to other countries. It looks like the Pakistani portions of the Asian Highway Network have yet to be mapped to the extent that its members have in neighboring countries. For reference, here is the Pakistan-China border in Americana at the moment, where you can see where the N-35 (Karakoram Highway in Pakistan) becomes the G314 (Karakoram Highway in China), but the AH4 default shield only shows on the Chinese side of the highway:
![image](https://user-images.githubusercontent.com/100172442/173210727-e1af77f3-80e4-4395-8654-84c2e51f4b0d.png)

**Comment by bgo-eiu** [comment]
I recall Pakistan's road route relations being tagged with a format like `network=PK:national` for National Highways, for example, but I will update this comment to reflect any other pertinent details when OSM comes back from the current downtime.

**Comment by 1ec5** [comment]
Yes, [`network=PK:national`](http://taginfo.geofabrik.de/asia/pakistan/tags/network=PK:national) is the tagging scheme for national routes in Pakistan. No other `PK:`-prefixed `network` value appears to be in use so far.

**Comment by bgo-eiu** [comment]
According to the Pakistan taginfo (helpful link, I forgot about the country projects), there are 28 relations with `network=PK:national` but 100 road route relations, including several from the Motorway network and provincial network, so it's likely many of them just lack a `network` tag. I could ask the Pakistan highway mappers if `network=PK:motorway`, `network=PK:Punjab` or some other tagging scheme sounds reasonable (shields aside, it would be nice to be able to distinguish between the three different networks which use `S` refs more easily).

**Comment by claysmalley** [comment]
The artwork for National Highway shields is already in the repo as the Japanese prefectural route shield, but should probably be renamed to something more generic:

![](https://raw.githubusercontent.com/ZeLonewolf/openstreetmap-americana/main/style/icons/shield40_jp_prefectural.svg)

**Comment by bgo-eiu** [comment]
That would work great and makes this easier!

Most of the road route relations in Pakistan (and the `network=PK:national` tags) were added over a decade ago by users that are no longer active. The more recent ones tend to have no `network` tag; those ones are also largely outside of the national network. I started using `network=PK:motorway` on the motorway network and added `network=PK:Punjab:LRR` for Lahore Ring Road. Those 3 shields would be straightforward to implement I think.

It didn't look like I would be stepping on any toes, but I did contact the one currently active Pakistan highway route mapper just in case. They've added routes like the [Gwadar East Bay Expressway](https://en.wikipedia.org/wiki/Gwadar_East_Bay_Expressway) which just opened a week ago and doesn't fit into any of the networks described above, which is pretty interesting.

**Comment by ZeLonewolf** [comment]
Sounds like a `blue_hexagon` to me...

**Comment by bgo-eiu** [comment]
![image](https://user-images.githubusercontent.com/100172442/173728668-5269eb79-ecaf-4597-9eb4-02f0f9fcc0ce.png)

Found on this archive of Pakistan sign codes: https://github.com/govdotpk/highway-code-book

I'm curious to find any photo examples of the signs on the right

**Comment by claysmalley** [comment]
National highways and motorways merged in #415.
